hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStore.java

```
@@ -213,6 +214,7 @@ public void close() throws IOException {
 
     RDBMetrics.unRegister();
     checkPointManager.close();
+    IOUtils.closeQuietly(rocksDBCheckpointDiffer);
```

qq: why do we want to swallow the exception here?

---

I don't think it throws any exception.

---

Sorry If I was not clear. I meant [IOUtils.closeQuietly()](https://github.com/apache/ozone/blob/0f1ba3f8a38e0d51bc402284607ac73fb1c116f9/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/IOUtils.java#L79) swallows the exception.
Why didn't we use `rocksDBCheckpointDiffer.close()` directly?

---

Because `rocksDBCheckpointDiffer` may be `null` (see 1c138a3).

---

Ah I see. Thanks.